src/fsharp/NameResolution.fs

```diff
@@ -1538,11 +1538,15 @@ type TcResultsSinkImpl(g, ?source: string) =
     let formatStringCheckContext =
         lazy
             source |> Option.map (fun source ->
-                let source = source.Replace("\r\n", "\n").Replace("\r", "\n")
                 let positions =
-                    source.Split('\n')
-                    |> Seq.map (fun s -> String.length s + 1)
-                    |> Seq.scan (+) 0
+                    seq {
```

Comment:
Instead of seq computation expressions and Seq.toArray you may just use the array computation expressions